aydin/nn/models/torch/test/test_torch_models.py

```diff
@@ -0,0 +1,137 @@
+from aydin.io.datasets import lizard, add_noise, camera, normalise
+from aydin.nn.models.torch.torch_res_unet import ResidualUNetModel
+from aydin.nn.models.torch.torch_unet import UNetModel, n2t_train, n2s_train
+
+
+def test_supervised_2D():
+    input_array = torch.zeros((1, 1, 64, 64))
+    model2d = ResidualUNetModel(
+        # (64, 64, 1),
+        nb_unet_levels=2,
+        spacetime_ndim=2,
+    )
+    result = model2d(input_array)
+    assert result.shape == input_array.shape
+    assert result.dtype == input_array.dtype
+
+
+def test_supervised_2D_n2t():
+    lizard_image = normalise(camera())
+    lizard_image = numpy.expand_dims(lizard_image, axis=0)
+    lizard_image = numpy.expand_dims(lizard_image, axis=0)
+
+    input_image = add_noise(lizard_image)
+
+    input_image = torch.tensor(input_image)
+    lizard_image = torch.tensor(lizard_image)
+
+    model = ResidualUNetModel(
+        nb_unet_levels=2,
+        spacetime_ndim=2,
+    )
+
+    n2t_train(input_image, lizard_image, model, nb_epochs=2)
+    result = model(input_image)
+
+    assert result.shape == input_image.shape
+    assert result.dtype == input_image.dtype
+
+
+def test_supervised_2D_n2s():
+    lizard_image = normalise(camera())
+    lizard_image = numpy.expand_dims(lizard_image, axis=0)
+    lizard_image = numpy.expand_dims(lizard_image, axis=0)
+
+    input_image = add_noise(lizard_image)
+
+    input_image = torch.tensor(input_image)
+
+    model = UNetModel(
+        nb_unet_levels=2,
+        spacetime_ndim=2,
+    )
+
+    n2s_train(input_image, model, nb_epochs=2)
+    model.cpu()
+    result = model(input_image)
+
+    assert result.shape == input_image.shape
+    assert result.dtype == input_image.dtype
+
+
+def test_masking_2D():
+    input_array = torch.zeros((1, 1, 64, 64))
+    model2d = UNetModel(
+        # (64, 64, 1),
+        nb_unet_levels=2,
+        spacetime_ndim=2,
+    )
+    result = model2d(input_array)
+    assert result.shape == input_array.shape
+    assert result.dtype == input_array.dtype
+
+
+# def test_jinet_2D():
+#     input_array = torch.zeros((1, 1, 64, 64))
+#     model2d = JINetModel((64, 64, 1), spacetime_ndim=2)
+#     result = model2d.predict([input_array])
+#     assert result.shape == input_array.shape
+#     assert result.dtype == input_array.dtype
+
+
+def test_supervised_3D():
+    input_array = torch.zeros((1, 1, 64, 64, 64))
+    model3d = UNetModel(
+        # (64, 64, 64, 1),
+        nb_unet_levels=2,
+        spacetime_ndim=3,
+    )
+    result = model3d(input_array)
+    assert result.shape == input_array.shape
+    assert result.dtype == input_array.dtype
+
+
+def test_masking_3D():
+    input_array = torch.zeros((1, 1, 64, 64, 64))
+    model3d = UNetModel(
+        # (64, 64, 64, 1),
+        nb_unet_levels=2,
+        spacetime_ndim=3,
+    )
+    result = model3d(input_array)
+    assert result.shape == input_array.shape
+    assert result.dtype == input_array.dtype
+
+
+# def test_various_masking_3D():
```

Comment:
Just a reminder that you can delete the comments if there's no plan to reactivate them.